rabbit/publisherpool.go

```diff
@@ -71,6 +71,18 @@ func (p *PublisherPool) Publish(ctx context.Context, msg publisher.Message) (err
 
 	span.RecordMetric(o11y.Timing("pool.publish", "exchange", "key", "content_type"))
 
+	return p.publishMandatory(ctx, msg)
+}
+
+func (p *PublisherPool) PublishOptional(ctx context.Context, msg publisher.Message) (err error) {
```

Comment:
Can we add a not here that this should be used while new message is being setup?

Reply:
![](https://i.pinimg.com/originals/a8/30/37/a8303735da0a7272bf11a762979dca4a.png)